### PR TITLE
Add `part_of` to country data

### PR DESCRIPTION
**Pull request by Skarvion**
Continuing from #45 

I've added `part_of` flag that points to the country code that "owns" it. 

Honestly, it's a very murky area. There are many different kinds of administrative privilege. From simple ones such as French Territories, to the blurry ones such as British Crown Dependencies that are governmentally independent and also internationally recognised as separate entities, yet still have the British monarch as the head of state. Also, Greenland is an autonomous territory of Denmark.

But for the sake of cleaning up the data, I think what we have now is good. If there's any dispute about whether the `part_of` flag was added correctly, can have a discussion again.

I've added test cases that guarantees the code of primary countries are always unique, and all `part_of` points to existing primary country codes.

And there are also typo fixes for American Samoa, Vietnam, and Western Sahara. There are countries with outdated name such as North Macedonia but still referred to as "Former Yugoslav State", but will fix it later.

### CODE REVIEW

**Comment by necessarylion** [comment]
@Skarvion Just wondering, are there any territories part of two or more countries?
I am wondering because may be better to use array if there is any? 

Example
```ts
{ name: 'Antarctica', dial_code: '+672', code: 'AQ', flag: '🇦🇶', part_of: ['US', 'GB'] },
```

**Comment by Skarvion** [comment]
From data structure point of view, I think it's good to have it as an array. So we can have the flexibility.

But in practice, it's a muddy area when talking about who owns what. In Antartica example, while different nations put bases and "claim" parts of Antartica, technically speaking no single country owns Antartica following the Antartic Treaty. So those claims are unofficial.

Another example is Western Sahara that is heavily disputed between Morocco and Sahrawi. Yet it has its own ISO code, I don't know why. To say that it's owned by both countries will spark debate and definitely not worth the trouble.

I can convert `part_of` to an array of string, but we still use the data that we have now. So we start with an array of single code. If there's any more changes, then that'll be for later. Is that okay?

**Comment by necessarylion** [comment]
> I can convert part_of to an array of string, but we still use the data that we have now. So we start with an array of single code. If there's any more changes, then that'll be for later. Is that okay?

Yeah. Sound good for me.

**Comment by Skarvion** [comment]
I have force pushed new change to the code, `part_of` uses array now

File: src/country.ts [hunk 1]
```diff
@@ -48,6 +48,10 @@ class Country {
     return this.data.country_code ?? this.data.dial_code
   }
 
+  public get partOf(): string[] | null {
```
Comment (necessarylion) [inline src/country.ts:51]: Shall we return empty array when not defined?

Comment (Skarvion) [reply src/country.ts]: Fixed

**Review by necessarylion — APPROVED** [review]

**Comment by necessarylion** [comment]
Thanks for the PR, I will include in next release.

**Pull request merged**